lib/cylc/config.py

```diff
@@ -274,6 +274,21 @@ def __init__(self, suite, fpath, template_vars=[],
         else:
             self.initial_point.standardise()
 
+        # Validate initial cycle point against any constraints
+        if self.cfg['scheduling']['initial cycle point constraints']:
```

Comment:
This logical test is not needed as the config item evaluates to an empty list rather than None, if no constraints are supplied.